Make consistent the expectation around non-existent fragments directory (#557)
twisted/towncrier: docs/cli.rst

Build the combined news file from news fragments.
``build`` is also assumed if no command is passed.

If there are no news fragments (including an empty fragments directory or a
non-existent directory), a notice of "no significant changes" will be added to
the news file.

> By default, the processed news fragments are removed using ``git``, which will
> also remove the fragments directory if now empty.

Reviewer: ```suggestion
By default, the processed news fragments are removed using ``git rm``.
```

I don't think that `towncrier` (or git) explicitly removes the empty dir.

Author: I tested this expectation in one of my projects. For example, you can checkout /jaraco/skeleton, which currently has one fragment, and run towncrier build 0.0.0, and it will remove the newsfragmenta dir.

Author: Here's where I did just that yesterday:

```
 @ work-on skeleton
 skeleton main @ towncrier build --version 0.0.0
Loading template...
Finding news fragments...
Rendering news fragments...
Writing to newsfile...
Staging newsfile...
I want to remove the following files:
/Users/jaraco/code/jaraco/skeleton/newsfragments/+drop-py37.feature.rst
Is it okay if I remove those files? [Y/n]: y
Removing news fragments...
Done!
 skeleton main @ cat NEWS.rst
0.0.0
=====

Features
--------

- Require Python 3.8 or later.
 skeleton main @ towncrier build --version 0.0.1
Loading template...
Finding news fragments...
Failed to list the news fragment files.
FileNotFoundError: [Errno 2] No such file or directory: '/Users/jaraco/code/jaraco/skeleton/newsfragments'

 skeleton main @ mkdir newsfragments
 skeleton main @ towncrier build --version 0.0.1
Loading template...
Finding news fragments...
Rendering news fragments...
Writing to newsfile...
Staging newsfile...
No news fragments to remove. Skipping!
Done!
```
